pkgs/development/python-modules/piano-transcription-inference/default.nix

```diff
@@ -0,0 +1,64 @@
+{ stdenv
+, lib
+, buildPythonPackage
+, fetchPypi
+, fetchpatch
+, fetchurl
+, matplotlib
+, mido
+, librosa
+, torchlibrosa
+}:
+
+buildPythonPackage rec {
+  pname = "piano-transcription-inference";
+  version = "0.0.5";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-nbhuSkXuWrekFxwdNHaspuag+3K1cKwq90IpATBpWPY=";
+  };
+
+  checkpoint = fetchurl {
+    name = "piano-transcription-inference.pth";
+    url = "https://zenodo.org/record/4034264/files/CRNN_note_F1%3D0.9677_pedal_F1%3D0.9186.pth?download=1";
+    sha256 = "sha256-w/qXMHJb9Kdi8cFLyAzVmG6s2gGwJvWkolJc1geHYUE=";
+  };
+
+  propagatedBuildInputs = [
+    matplotlib
+    mido
+    librosa
+    torchlibrosa
+  ];
+
+  patches = [
+    # Fix run against librosa 0.9.0
+    # https://github.com/qiuqiangkong/piano_transcription_inference/pull/10
+    (fetchpatch {
+      url = "https://github.com/qiuqiangkong/piano_transcription_inference/commit/b2d448916be771cd228f709c23c474942008e3e8.patch";
+      sha256 = "sha256-8O4VtFij//k3fhcbMRz4J8Iz4AdOPLkuk3UTxuCSy8U=";
+    })
+  ];
+
+  postPatch = ''
+    substituteInPlace piano_transcription_inference/inference.py --replace \
+      "checkpoint_path='{}/piano_transcription_inference_data/note_F1=0.9677_pedal_F1=0.9186.pth'.format(str(Path.home()))" \
+      "checkpoint_path='$out/share/checkpoint.pth'"
+  '';
+
+  postInstall = ''
+    mkdir "$out/share"
+    cp "${checkpoint}" "$out/share/checkpoint.pth"
+  '';
+
+  # Project has no tests
+  doCheck = false;
```

Comment:
Missing pythonImportsCheck.

Reply:
It's a bit tricky, `torchlibrosa` `piano-transcription-inference` `pianotrans` all depend on `pytorch`, which is not always available on all platform by default, simply use pythonImportsCheck to test will give error: `ModuleNotFoundError: No module named 'torch'`

My solution to make the final `pianotrans` command work is to choose `pytorch` or `pytorch-bin` based on host platforms, but I don't think it's good idea to put this logic to the package definitions.

Reply:
If you can't import the module the package is incomplete and this is an error which needs to be fixed.

> which is not always available on all platform by default

Then the package is restricted to the working platforms. Not like you could use it on the other platforms anyway. Or we just use pytorch-bin on all platforms.

Reply:
All these packages `torchlibrosa` `piano-transcription-inference` `pianotrans` work on all 4 platforms, tested using VMs.

The problem is `pytorch`(build from sources) and `pytorch-bin`(build from binaries) on nixpkgs are both incomplete.

`pytorch-bin` on nixpkgs lacks aarch64-linux, but pytorch upsteam do support aarch64-linux, see https://download.pytorch.org/whl/torch_stable.html

`pytorch` on nixpkgs marked as broken on darwin, I've tried to fix it, https://github.com/azuwis/nixpkgs/commit/17915c9feb0b46fd2cf44f11987d1d4a485932e6 tested on aarch64-darwin, built fine, but running program against it is much slower compared to `pytorch-bin`.

`pytorch` and `pytorch-bin` are interchangeable, how about we choose the right one for end user, in `pkgs/top-level/python-packages.nix`:
```
  pytorch-src = callPackage ../development/python-modules/pytorch {
    cudaSupport = pkgs.config.cudaSupport or false;
  };

  pytorch-bin = callPackage ../development/python-modules/pytorch/bin.nix { };

  pytorch = if stdenv.isDarwin then pytorch-bin else pytorch-src;
```

Reply:
I am not sure if we want to do that. @jonringer do you have an opinion on that?

Otherwise I would just do it for the packages right now.